charts/crossplane-azure-upbound/templates/providers.yaml

```
@@ -0,0 +1,24 @@
+{{- if and .Values.global.enabled_azure_upbound .Values.provider.enabled }}
+{{- range .Values.providers }}
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-azure-{{ . }}
+  annotations:
+    argocd.argoproj.io/sync-wave: "3"
+    {{- range $key, $value := $.Values.provider.metadata.annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- if $.Values.provider.metadata.labels }}
+  labels:
+    {{- range $key, $value := $.Values.provider.metadata.labels }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
+spec:
+  package: {{ $.Values.provider.package.registry }}/provider-azure-{{ . }}:{{ $.Values.provider.package.version }}
```

```suggestion
  package: {{ $.Values.provider.package.registry }}/provider-azure-{{ . }}:{{ $.Values.provider.package.version }}
  runtimeConfigRef:
    apiVersion: pkg.crossplane.io/v1beta1
    kind: DeploymentRuntimeConfig
    name: {{ $.Values.deploymentRuntimeConfig.metadata.name }}
```

You are missing the reference to the deployment runtime config that you are creating